### ISSUE TITLE
Creating model with nested models including nil values

### ISSUE THREAD

**Issue by ehannes** [my-codeworks/fortnox-api#62]
When creating a model, attributes not provided will have nil as value. When corresponding repository tries to save the model, it should remove nil values. `Repository::JSONConversion.remove_nil_values( hash )` manages this, but it does not handle nested hashes yet.

It is defined as
```
def remove_nil_values( hash )
  hash.delete_if{ |_, value| value.nil? }
end
```

Ex:
```
irb(main):054:0> order = {:order=>{:order_rows=>{"1"=>{"price"=>0, "ordered_quantity"=>0, "article_numer"=>"0000", "bar"=>nil}}}, :foo=>nil}
=> {:order=>{:order_rows=>{"1"=>{"price"=>0, "ordered_quantity"=>0, "article_numer"=>"0000", "bar"=>nil}}}, :foo=>nil}
irb(main):055:0> order.delete_if{ |_, value| value.nil? }
=> {:order=>{:order_rows=>{"1"=>{"price"=>0, "ordered_quantity"=>0, "article_numer"=>"0000", "bar"=>nil}}}}
```
`:foo` is correctly removed, but `:bar` is not.

**Comment by ehannes** [comment]
Problem with removing nil values before save is solved in b58bf004ce2d8900e6ac18e57817496c4bf74063

The next problem is that we need to tell the repository how to convert the nested model to into JSON. If I try to save an `Order` with a a price, the gem sends a request to Fortnox with the following JSON:
`"OrderRows":[{"price":100.0},{"price":101.5}]`. The problem here is that `price` should be capitalized.

**Comment by d-Pixie** [comment]
You should just delegate all of this to that model. So that model is responsible for returning it's own hash for example, or filtering it's own keys etc etc. The parent model should only know how to send the message on to any children, not how to actually convert the children into hashes or how to filter their keys.

**Comment by d-Pixie** [comment]
@ehannes this one should also be fixed by #82 , right?

**Comment by ehannes** [comment]
Yes, see the reference above.
